taskchampion/src/errors.rs

```diff
@@ -45,4 +43,41 @@ other_error!(google_cloud_storage::http::Error);
 #[cfg(feature = "server-gcp")]
 other_error!(google_cloud_storage::client::google_cloud_auth::error::Error);
 
+/// Convert ureq errors more carefully
+#[cfg(feature = "server-sync")]
+impl From<ureq::Error> for Error {
+    fn from(ureq_err: ureq::Error) -> Self {
+        match ureq_err {
+            ureq::Error::Status(status, response) => {
+                let msg = format!(
+                    "{} responded with {} {}",
+                    response.get_url(),
+                    status,
+                    response.status_text()
+                );
+                Self::Server(msg)
+            }
+            ureq::Error::Transport(_) => Self::Server(ureq_err.to_string()),
```

Comment:
It might be nice to add a test for the `Transport` case as well since we're introducing the possibility of a mistake here and these errors might be so unusual and non-deterministic that a mishandling would result in difficult-to-diagnose "phantom bugs".

Reply:
True, but quite a bit harder as those are not constructible (no new method)